src/web/components/SignInGate/displayRule.ts

```diff
@@ -99,68 +97,28 @@ export const canShow = (
 	);
 };
 
-const allCmpConsents: (
-	| 'a9'
-	| 'acast'
-	| 'braze'
-	| 'comscore'
-	| 'facebook-mobile'
-	| 'fb'
-	| 'firebase'
-	| 'google-analytics'
-	| 'google-mobile-ads'
-	| 'google-sign-in'
-	| 'google-tag-manager'
-	| 'googletag'
-	| 'ias'
-	| 'inizio'
-	| 'ipsos'
-	| 'lotame'
-	| 'nielsen'
-	| 'ophan'
-	| 'permutive'
-	| 'prebid'
-	| 'redplanet'
-	| 'remarketing'
-	| 'sentry'
-	| 'teads'
-	| 'twitter'
-	| 'youtube-player'
-)[] = [
-	'a9',
-	'acast',
-	'braze',
-	'comscore',
-	'fb',
-	'google-analytics',
-	'google-mobile-ads',
-	'google-tag-manager',
-	'googletag',
-	'ias',
-	'inizio',
-	'ipsos',
-	'lotame',
-	'nielsen',
-	'ophan',
-	'permutive',
-	'prebid',
-	'redplanet',
-	'remarketing',
-	'sentry',
-	'teads',
-	'twitter',
-	'youtube-player',
-];
-
 export const hasRequiredConsents = (): Promise<boolean> => {
+	const hasConsentedToAll = (state: ConsentState) => {
+		const consentFlags = state.tcfv2?.consents
+			? Object.values(state.tcfv2.consents)
+			: [];
+		const vendorConsentFlags = state.tcfv2?.vendorConsents
+			? Object.values(state.tcfv2.vendorConsents)
+			: [];
+		const isEmpty =
+			consentFlags.length === 0 || vendorConsentFlags.length === 0;
+
+		return (
+			!isEmpty &&
+			[...consentFlags, ...vendorConsentFlags].every(
+				(hasConsented) => hasConsented,
+			)
```

Comment:
FWIW [we’ve used][] and liked `.every(Boolean)`. So did [membership][].

[we’ve used]: https://github.com/guardian/frontend/blob/89802f2230100a9b9c1dc42e9ff4c64f77ec6e3f/static/src/javascripts/projects/common/modules/commercial/build-page-targeting.js#L224
[membership]: https://github.com/guardian/membership-frontend/blob/43751df8baae6656cfd31fb6d695485c4ad3a19f/frontend/assets/javascripts/src/modules/analytics/cmp.es6#L27